version: "2"
services:
   namenode:
      image: apache/hadoop:2.10.2
      hostname: namenode
      command: ["hdfs", "namenode"]
      environment:
        ENSURE_NAMENODE_DIR: "/tmp/hadoop-root/dfs/name"
        HADOOP_HOME: /opt/hadoop
      ports:
        - 9870:9870
      env_file:
        - ./config
      volumes:
        - /Users/xaohii/data/docker_local/hadoop/namenode_data:/tmp/hadoop-hadoop
      networks:
        - big_data-network
   datanode:
      image: apache/hadoop:2.10.2
      command: ["hdfs", "datanode"]
      environment:
        HADOOP_HOME: /opt/hadoop
      env_file:
        - ./config
      volumes:
        - /Users/xaohii/data/docker_local/hadoop/datanode_data:/tmp/hadoop-hadoop
      networks:
        - big_data-network
   resourcemanager:
      image: apache/hadoop:2.10.2
      hostname: resourcemanager
      command: ["yarn", "resourcemanager"]
      environment:
        HADOOP_HOME: /opt/hadoop
      ports:
         - 8088:8088
      env_file:
        - ./config
      networks:
        - big_data-network
   nodemanager:
      image: apache/hadoop:2.10.2
      command: ["yarn", "nodemanager"]
      depends_on:
        - resourcemanager
      environment:
        HADOOP_HOME: /opt/hadoop
      env_file:
        - ./config
      volumes:
        - /Users/xaohii/data/docker_local/share:/share
      networks:
        - big_data-network
networks:
  big_data-network:
    external: true